test/remote/too_many_account_status_check.js

```diff
@@ -0,0 +1,136 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+var test = require('tap').test
+var TestServer = require('../test_server')
+var Promise = require('bluebird')
+var restify = Promise.promisifyAll(require('restify'))
+var mcHelper = require('../memcache-helper')
+
+var TEST_IP = '192.0.2.1'
+
+var config = {
+  listen: {
+    port: 7000
+  }
+}
+
+// Override limit values for testing
+process.env.MAX_ACCOUNT_STATUS_CHECK = 2
+process.env.RATE_LIMIT_INTERVAL_SECONDS = 1
+
+var testServer = new TestServer(config)
+
+var client = restify.createJsonClient({
+  url: 'http://127.0.0.1:' + config.listen.port
+})
+
+Promise.promisifyAll(client)
+
+function delay(ms) {
+  var deferred = Promise.pending()
+  setTimeout(function(){
+    deferred.resolve()
+  }, ms)
+  return deferred.promise
+}
```

Comment:
We should be able to use [1] directly here.

[1] http://bluebirdjs.com/docs/api/delay.html